api/renter.go

```diff
@@ -350,49 +387,141 @@ func (api *API) renterDeleteHandler(w http.ResponseWriter, req *http.Request, ps
 
 // renterDownloadHandler handles the API call to download a file.
 func (api *API) renterDownloadHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	destination := req.FormValue("destination")
-	// Check that the destination path is absolute.
-	if !filepath.IsAbs(destination) {
-		WriteError(w, Error{"destination must be an absolute path"}, http.StatusBadRequest)
+	p, errmsg := api.parseAndValidateDownloadParameters(w, req, ps)
```

Comment:
`p` is not a typical name, would prefer the name `params` which is more commonly used throughout the project.